### DISCUSSION TITLE
Threding in Button Commands not working

### DISCUSSION THREAD

**Discussion by svn-josh** [TomSchimansky/CustomTkinter#611] [General]
Hello there,

```
self.button_1 = customtkinter.CTkButton(master=self.frame_left,
                                                text="Überprüfung",
                                                command=threading.Thread(target=self.start_module(fcheck)).start())
self.button_1.grid(row=2, column=0, pady=15, padx=20)
```

Using therding in commands will result them to be runned on startup (not on button press)
kinda fixed it by using an additional def. idk. if this is a good thing too
```

self.button_1 = customtkinter.CTkButton(master=self.frame_left,
                                                text="Überprüfung",
                                                command=self.b_startup)
self.button_1.grid(row=2, column=0, pady=15, padx=20)

def b_startup(self):
        threading.Thread(target=self.start_module(fcheck)).start()
```
but then when i run the command it kinda works, but the programm is getting no response while the action, so the threading must be broken then i guess

anyone got a better solution?

**Comment by Evolution0** [comment]
I ran into a similar issue and I what I did was use a lambda:
```python
self.button_1 = customtkinter.CTkButton(master=self.frame_left,
                                                text="Überprüfung",
                                                command=lambda x: threading.Thread(target=self.start_module(fcheck)).start())
self.button_1.grid(row=2, column=0, pady=15, padx=20)
```

Might work might not, its a general Tkinter thing rather than an issue with CustomTkinter specifically. Here is some info on why lambda usually works: https://stackoverflow.com/a/61156251

**Reply by svn-josh** [reply]
Kinda works, but the Main Programm is sill getting no response (timeout)

**Comment by sryu1** [comment]
I may be wrong but can you try it with removing the () after start?

**Reply by svn-josh** [reply]
yeah could, but sadly it does not fix my problem

**Comment by svn-josh** [comment]
so i am currently stuck with...

This works:
`self.button = customtkinter.CTkButton(master=self, command=threading.Thread(target=self.button_callback).start, text="Insert")`

But as i ad () to button_callback it does not work anymore:
`self.button = customtkinter.CTkButton(master=self, command=lambda:threading.Thread(target=self.button_callback('test')).start, text="Insert")`

As i see this i cant to anything to use threding with customtkinter? @TomSchimansky

**Comment by Akascape** [comment]
@GermanYoshi Try this: `command=lambda:threading.Thread(target=run_command, args=[]).start()`

**Reply by svn-josh** [reply]
There is no Error. I worte this small example App to show to problem.

https://pastebin.com/wDNvqxhL

**Reply by svn-josh** [reply]
The Goal is to run some kind of task that takes some time to execute and keep the App Updating itself.
(Show in the Example by the Progress bar)

**Reply by Akascape** [reply]
@GermanYoshi Ok, I see that you are using arguments directly with the threading function *button callback* which is causing this problem. You have to use `args` separately with threading:
```python
command=lambda: threading.Thread(target=self.button_callback, args=["test2"]).start()
```
This will solve the issue!

**Reply by svn-josh** [reply]
Thank you that worked :)
